graph/vertex/impl/AttributeVertexImpl.java

```diff
@@ -524,4 +555,47 @@ public DateTime asDateTime() {
             }
         }
     }
+
+    static class Value<VALUE> extends AttributeVertexImpl.Read<VALUE> implements AttributeVertex.Value<VALUE> {
+
+        private final AttributeVertex<VALUE> attributeVertex;
+
+        Value(AttributeVertex<VALUE> attributeVertex) {
+            super(attributeVertex.graph(), attributeVertex.iid());
+            this.attributeVertex = attributeVertex;
+        }
+
+        @Override
+        public AttributeVertex<VALUE> toAttribute() {
+            return attributeVertex;
+        }
+
+        @Override
+        public boolean isValue() {
+            return true;
+        }
+
+        @Override
+        public Value<VALUE> asValue() {
+            return this;
+        }
+
+        @Override
+        public Value<VALUE> toValue() {
+            return this;
+        }
+
+        @Override
+        public int compareTo(Vertex<?, ?> other) {
+            if (other.isThing() && other.asThing().isAttribute() && other.asThing().asAttribute().isValue()) {
+                Value<?> value = other.asThing().asAttribute().asValue();
+                int comp = compareValue(value);
+                return comp == 0 ? type().compareTo(value.type()) : comp;
+            } else return super.compareTo(other);
+        }
+
+        private <T> int compareValue(Value<T> valueVertex) {
```

Comment:
using this method allows us to insert a generic parameter for the `valueVertex`, giving us typed return values for its methods